src/main/java/seedu/address/commons/util/TimeUtil.java

```diff
@@ -0,0 +1,109 @@
+package seedu.address.commons.util;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Locale;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+/**
+ * Provides utility functions to handle and format time-related inputs.
+ * <p>
+ * This utility class provides multiple date and time formats for parsing,
+ * which can be helpful in accommodating various user inputs.
+ * </p>
+ */
+public class TimeUtil {
+    private static final DateTimeFormatter[] VALID_DATE_FORMATS = {
+        DateTimeFormatter.ISO_LOCAL_DATE,
+        DateTimeFormatter.BASIC_ISO_DATE,
+        DateTimeFormatter.ofPattern("d MMM yyyy", Locale.ENGLISH),
+        DateTimeFormatter.ofPattern("d MMMM yyyy", Locale.ENGLISH)
```

Comment:
Should this be `dd MM yyyy` instead?

Also, suggestion to stick with 1 `DateTime` format to keep it simple, rather than multiple date formats that can make documentation/usage confusing. Thoughts?

Reply:
These are the [examples](https://github.com/AY2324S1-CS2103T-F08-3/tp/pull/51/files#diff-f556a56cd1d8547670c2463c6592c9b62ba5b6e59874a65ce1e18fe3ed8339a9R84-R89) of the valid Date formats currently accepted. `dd MM yyyy` would correspond to `28 09 2023`. I am more of the view that having more date formats improve user experience, rather than causing documentation/usage to be confusing. Besides, I provided a helper function [TimeUtil::getValidDateFormats](https://github.com/AY2324S1-CS2103T-F08-3/tp/pull/51/files#diff-920d90aa208aad607eed72a09c7678ce913ebd69994467b2fe30b93a55bed5b7R70-R84) that can be called to provide examples of valid date formats to the user.

Reply:
That sounds fine, you brought up a good point. Let's stick with the current implementation!